pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -265,3 +279,22 @@ func (r *ReconcileNodeMaintenance) StartPodInformer(node *corev1.Node, stop <-ch
 
 	return nil
 }
+
+func (r *ReconcileNodeMaintenance) reconcileAndError(nm *kubevirtv1alpha1.NodeMaintenance, result reconcile.Result, err error) (reconcile.Result, error) {
+	//should always be false by default unless completed
+	nm.Status.Phase = kubevirtv1alpha1.MaintenanceFailed
```

Comment:
Phase=Success|Failed need to be finalized states.

Is it possible for the phase to be set to Failed here yet at some point get retied and later on get set to Success? If so, that needs to change.

Conditions exist to represent these transient states where a CR might be in the Running Phase yet be temporarily failing on a transient error that will result in a retry later on.

Reply:
@davidvossel  good point.

> Is it possible for the phase to be set to Failed here yet at some point get retied and later on get set to Success?

Yes, it is possible with the current implementation. Since there is no timeout/time buffer field - we do not have a finalized state for failed. the controller will just keep trying to reconcile and process the maintenance CR until its deleted.

> Conditions exist to represent these transient states where a CR might be in the Running Phase yet be temporarily failing on a transient error that will result in a retry later on.

Conditions do look like the better way to express the transient errors if there is no refrain of still using them (also in future versions)
If so, then i would remove the LastError field that states the last transient error and replace it with conditions.

Reply:
@davidvossel  Fail status now removed